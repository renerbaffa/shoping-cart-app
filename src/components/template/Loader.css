.Loader-container {
  position: absolute;
  margin-top: 90px;
  z-index: 1;
}
