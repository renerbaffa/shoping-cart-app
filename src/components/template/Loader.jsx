import React from 'react';
import PropTypes from 'prop-types';

import Spinner from 'react-loader-spinner'

import './Loader.css';

const MyLoader = ({ show, ...other }) => {
  if (!show) {
    return null;
  }

  return (
    <div className="Loader-container">
      <Spinner
        type="TailSpin"
        color="gray"
      />
    </div>
  );
}

MyLoader.propTypes = {
  show: PropTypes.bool,
};

MyLoader.defaultProps = {
  show: false,
};

export default MyLoader;
