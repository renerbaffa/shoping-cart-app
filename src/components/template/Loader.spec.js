import React from 'react';
import renderer from 'react-test-renderer';

import Loader from './Loader';

describe('<Loader />', () => {
  let  loader;

  it('should return null if show prop is false', () => {
    loader = renderer.create(<Loader />).toJSON();
    expect(loader).toMatchSnapshot();
  });

  it('should show spinner correctly', () => {
    loader = renderer.create(<Loader show />).toJSON();
    expect(loader).toMatchSnapshot();
  });
})