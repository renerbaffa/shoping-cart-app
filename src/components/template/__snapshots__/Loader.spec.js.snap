// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should return null if show prop is false 1`] = `null`;

exports[`<Loader /> should show spinner correctly 1`] = `
<div
  className="Loader-container"
>
  <div>
    <svg
      height={80}
      viewBox="0 0 38 38"
      width={80}
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs>
        <linearGradient
          id="a"
          x1="8.042%"
          x2="65.682%"
          y1="0%"
          y2="23.865%"
        >
          <stop
            offset="0%"
            stopColor="gray"
            stopOpacity="0"
          />
          <stop
            offset="63.146%"
            stopColor="gray"
            stopOpacity=".631"
          />
          <stop
            offset="100%"
            stopColor="gray"
          />
        </linearGradient>
      </defs>
      <g
        fill="none"
        fillRule="evenodd"
      >
        <g
          transform="translate(1 1)"
        >
          <path
            d="M36 18c0-9.94-8.06-18-18-18"
            id="Oval-2"
            stroke="url(#a)"
            strokeWidth="2"
          >
            <animateTransform
              attributeName="transform"
              dur="0.9s"
              from="0 18 18"
              repeatCount="indefinite"
              to="360 18 18"
              type="rotate"
            />
          </path>
          <circle
            cx="36"
            cy="18"
            fill="#fff"
            r="1"
          >
            <animateTransform
              attributeName="transform"
              dur="0.9s"
              from="0 18 18"
              repeatCount="indefinite"
              to="360 18 18"
              type="rotate"
            />
          </circle>
        </g>
      </g>
    </svg>
  </div>
</div>
`;
